controller/api/destination/watcher/ip_watcher.go

```diff
@@ -146,11 +169,37 @@ func (iw *IPWatcher) GetSvcID(clusterIP string) (*ServiceID, error) {
 }
 
 // GetPod returns the pod that corresponds to an IP address if one exists.
-func (iw *IPWatcher) GetPod(podIP string) (*corev1.Pod, error) {
-	objs, err := iw.k8sAPI.Pod().Informer().GetIndexer().ByIndex(podIPIndex, podIP)
+func (iw *IPWatcher) GetPod(podIP string, port uint32) (*corev1.Pod, error) {
+	addr := fmt.Sprintf("%s:%d", podIP, port)
+	hostIPPods, err := iw.getIndexedPods(hostIPIndex, addr)
+	if err != nil {
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+	// If there is exactly 1 pod found then it is returned. More than 1 pod is
+	// not handled because that would violate the k8s contract of unique
+	// hostIP:hostPort addresses.
+	if len(hostIPPods) == 1 {
+		return hostIPPods[0], nil
+	}
```

Comment:
Regardless of which approach we go with, we should add a comment here explaining what we can safely assume at this point (e.g. do we know if the IP is on the pod or host network? etc.)